core/infra/repository/world_state_repository_with_level_db.cpp

```diff
@@ -7,82 +7,99 @@
-          leveldb::Options options;
-          options.error_if_exists = false;        
-          options.create_if_missing = true;
-          loggerStatus(leveldb::DB::Open(options, "/tmp/irohaDB", &tmpDb)); //TODO: This path should be configurable
-          db.reset(tmpDb);
-      }
+      bool add(const std::string &key, const std::string &value) {
+          if (detail::db == nullptr) detail::loadDb();
 
-      bool add(const std::string& key, const std::string& value){
-          return loggerStatus( db->Put(leveldb::WriteOptions(), key, value));
+          return detail::loggerStatus(detail::db->Put(leveldb::WriteOptions(), key, value));
       }
 
-      bool update(const std::string& key, const std::string& value){
+      bool update(const std::string &key, const std::string &value) {
+          if (detail::db == nullptr) detail::loadDb();
+
           std::string dummy;
-          if( loggerStatus(db->Get(leveldb::ReadOptions(), key, &dummy)) ){
+          if (detail::loggerStatus(detail::db->Get(leveldb::ReadOptions(), key, &dummy))) {
               leveldb::WriteBatch batch;
               batch.Delete(key);
               batch.Put(key, value);
-              return loggerStatus( db->Write(leveldb::WriteOptions(), &batch));
+              return detail::loggerStatus(detail::db->Write(leveldb::WriteOptions(), &batch));
           }
+          return false;
       }
 
-      bool remove(const std::string& key){
-          return loggerStatus(db->Delete(leveldb::WriteOptions(), key));
+      bool remove(const std::string &key) {
+          if (detail::db == nullptr) detail::loadDb();
+          return detail::loggerStatus(detail::db->Delete(leveldb::WriteOptions(), key));
       }
 
-      std::string find(const std::string& key){
+      std::string find(const std::string &key) {
+          if (detail::db == nullptr) detail::loadDb();
+
           std::string readData;
-          loggerStatus(db->Get(leveldb::ReadOptions(), key, &readData));
-          if(readData != ""){
+          detail::loggerStatus(detail::db->Get(leveldb::ReadOptions(), key, &readData));
+          if (readData != "") {
               return readData;
           } else {
               return "";
           }
       }
 
       std::string findOrElse(
-        const std::string& key,
-        const std::string& defaultValue
+              const std::string &key,
+              const std::string &defaultValue
       ) {
+          if (detail::db == nullptr) detail::loadDb();
+
           std::string result = "";
-          loggerStatus(db->Get(leveldb::ReadOptions(), key, &result));          
-          if(result == "") {
+          detail::loggerStatus(detail::db->Get(leveldb::ReadOptions(), key, &result));
+          if (result == "") {
               return defaultValue;
           } else {
               return result;
           }
       }
 
-      bool isExist(const std::string& key){
+      bool isExist(const std::string &key) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The function 'isExist' is never used.](https://www.codacy.com/app/soramitsu/iroha/file/4256547628/issues/source?bid=3712501&fileBranchId=3831158#l99)